posthog/clickhouse/query_tagging.py

```diff
@@ -0,0 +1,23 @@
+# This module is responsible for adding tags/metadata to outgoing clickhouse queries in a thread-safe manner
+
+import threading
+
+thread_local_storage = threading.local()
+
+
+def get_query_tags():
+    try:
+        return thread_local_storage.query_tags
```

Comment:
A quick google suggests celery runs each task in its own thread and django runs each request in its own thread. So this seems safe

If we switch Django to ASGI at some point then this wouldn't be safe any more (I think) but that's a problem for the future :)